servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -0,0 +1,457 @@
+   * @throws IOException this exception is not thrown in the tiered block store implementation
+   */
+  public void accessBlock(long userId, long blockId) throws IOException {
+    mBlockStore.accessBlock(userId, blockId);
+  }
+
+  /**
+   * Add the checkpoint information of a file. The information is from the user <code>userId</code>.
+   *
+   * This method is normally triggered from {@link tachyon.client.FileOutStream#close()} if and only
+   * if {@link tachyon.client.WriteType#isThrough()} is true. The current implementation of
+   * checkpointing is that through {@link tachyon.client.WriteType} operations write to
+   * {@link tachyon.underfs.UnderFileSystem} on the client's write path, but under a user temp
+   * directory (temp directory is defined in the worker as {@link #getUserUfsTmpFolder(long)}).
+   *
+   * @param userId The user id of the client who sends the notification
+   * @param fileId The id of the checkpointed file
+   * @throws TException if the file does not exist or cannot be renamed
+   * @throws IOException if the update to the master fails
+   */
+  public void addCheckpoint(long userId, int fileId) throws TException, IOException {
+    // TODO This part needs to be changed.
+    String srcPath = CommonUtils.concatPath(getUserUfsTmpFolder(userId), fileId);
+    String ufsDataFolder =
+        mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER, Constants.DEFAULT_DATA_FOLDER);
+    String dstPath = CommonUtils.concatPath(ufsDataFolder, fileId);
+    try {
+      if (!mUfs.rename(srcPath, dstPath)) {
+        throw new FailedToCheckpointException("Failed to rename " + srcPath + " to " + dstPath);
+      }
+    } catch (IOException e) {
+      throw new FailedToCheckpointException("Failed to rename " + srcPath + " to " + dstPath);
+    }
+    long fileSize;
+    try {
+      fileSize = mUfs.getFileSize(dstPath);
+    } catch (IOException e) {
+      throw new FailedToCheckpointException("Failed to getFileSize " + dstPath);
+    }
+    mMasterClient.addCheckpoint(mWorkerId, fileId, fileSize, dstPath);
+  }
+
+  /**
+   * Cleans up after users, to prevent zombie users. This method is called periodically.
+   */
+  public void cleanupUsers() throws IOException {
+    for (long user : mUsers.getTimedOutUsers()) {
+      mUsers.removeUser(user);
+      mBlockStore.cleanupUser(user);
+    }
+  }
+
+  /**
+   * Commits a block to Tachyon managed space. The block must be temporary. The block is
+   * persisted after {@link BlockStore#commitBlock(long, long)}. The block will not be accessible
+   * until {@link MasterClient#worker_cacheBlock(long, long, long, long, long)}
+   * succeeds
+   *
+   * @param userId The id of the client
+   * @param blockId The id of the block to commit
+   * @return true if successful, false otherwise
+   * @throws IOException if the block to commit does not exist
+   */
+  // TODO: This may be better as void
+  public boolean commitBlock(long userId, long blockId) throws IOException {
```

Comment:
I think the boolean returned is not really used, given we are throwing exceptions.